org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/XMLExtensionsRegistry.java

```diff
@@ -197,6 +197,12 @@ void unregisterExtension(IXMLExtension extension) {
 		extensions.remove(extension);
 		extension.stop(this);
 	}
+	
+	public void disposeExtensions() {
+		for (IXMLExtension extension : extensions) {
+			unregisterExtension(extension);

```

Comment:
Not tested but are you sure it works since unregisterEXtension remove the extension? Please add tests which create XML extension registry add some extension and callb dispose. You could check the state of extension like stopped and check there are none extensions

Reply:
Ok no problem